stream/stream.go

```diff
@@ -85,25 +89,42 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	err = h.Searcher.StreamSearch(ctx, args.Q, args.Opts, SenderFunc(func(event *zoekt.SearchResult) {
-		// We don't want to send events over the wire if they just contain stats and no
-		// file matches. Hence, in case we didn't find any results, we will just
-		// aggregate the stats.
+		// We don't want to send events over the wire if they don't contain file
+		// matches. Hence, in case we didn't find any results, we aggregate the stats
+		// and send them out in regular intervals.
 		if len(event.Files) == 0 {
-			aggStats.Add(event.Stats)
+			aggCount++
+
+			agg.Stats.Add(event.Stats)
+			agg.Progress = event.Progress
+
+			if aggCount%100 == 0 && !agg.Stats.Zero() {
+				send(&agg)
+				agg = zoekt.SearchResult{}
+			}
+
 			return
 		}
 
 		// If we have aggregate stats, we merge them with the new event before sending
-		// it, and reset aggStats afterwards.
-		if !aggStats.Zero() {
-			defer func() { aggStats = zoekt.Stats{} }() // reset stats
-			event.Stats.Add(aggStats)
+		// it. We drop agg.Progress, because we assume that event.Progress reflects the
+		// latest status.
+		if !agg.Stats.Zero() {
+			event.Stats.Add(agg.Stats)
+			agg = zoekt.SearchResult{}
 		}
+
 		send(event)
 	}))
 
-	if err == nil && !aggStats.Zero() {
-		send(&zoekt.SearchResult{Stats: aggStats})
+	if err == nil && !agg.Stats.Zero() {
+		send(&zoekt.SearchResult{
+			Stats: agg.Stats,
+			Progress: zoekt.Progress{
+				Priority:           math.Inf(-1),
```

Comment:
this is new, is it safe? I'm not sure how this interacts.

Reply:
We already send `math.Inf(-1)` if we find a result in the very last shard we search. In `shards.go` we call `pending.max()` which will return `math.Inf(-1)` if there is no shard left to search.

Reply:
Edit: Just realised you are referring to `Priority` and not `MaxPendingPriority`. 

Just thinking about Zoekt, I think `Priority=match.Inf(-1)` makes sense because this event is the last `eventMatches` event we send and it comes without matches. Additionally, if we found matches in the last shard, we already communicated `MaxPendingPriority=math.Inf(-1)`, so sending `Priority=0` afterwards seems inconsistent. 

In Sourcegraph we will currently ignore `Progress.Priority` if the event has no matches.